### ISSUE TITLE
run_classifier_infer.py多次预测，输出结果不一致

### ISSUE THREAD

**Issue by 0x0004** [dbiir/UER-py#149]
数据集：
将datasets\douban_book_review\test.tsv和test_nolabel.tsv删减至只剩一行，或使用任意数据。
模型：
使用https://huggingface.co/uer/chinese_roberta_L-12_H-768 转换成uer可用。
命令行：
python inference/run_classifier_infer.py --load_model_path models/uer.bin --vocab_path models/google_zh_vocab.txt --test_path datasets/douban_book_review/test_nolabel.tsv --prediction_path datasets/douban_book_review/prediction.tsv --labels_num 2 --output_prob --embedding word_pos_seg --encoder transformer --mask fully_visible
结果：
1	0.4978545308113098 0.5021454691886902
label	prob
1	0.48731300234794617 0.5126870274543762
label	prob
0	0.5316037535667419 0.46839630603790283
label	prob
1	0.44275444746017456 0.5572454929351807
label	prob
1	0.482973575592041 0.5170264840126038
label	prob
0	0.5044168829917908 0.49558305740356445
label	prob
0	0.5012422204017639 0.4987577795982361
label	prob
1	0.46198830008506775 0.5380117297172546

遇到的问题：预测概率每次不相同。标签也不同。

**Comment by Embedding** [comment]
您好，https://huggingface.co/uer/chinese_roberta_L-12_H-768 是预训练模型，不是分类模型，不能直接使用 *python inference/run_classifier_infer.py*
建议先仔细阅读快速上手以及 https://huggingface.co/uer/ 中对预训练权重的描述

**Comment by 0x0004** [comment]
> 您好，https://huggingface.co/uer/chinese_roberta_L-12_H-768 是预训练模型，不是分类模型，不能直接使用 _python inference/run_classifier_infer.py_
> 建议先仔细阅读快速上手以及 https://huggingface.co/uer/ 中对预训练权重的描述

明白了，那么如果想取得最好的分类效果，是应该在chinese_roberta_L12H768上preprocess+pretrain去增量预训练，还是直接用run_classifier微调后输出模型。

**Comment by Embedding** [comment]
对于分类任务，如果计算资源比较充裕，可以加载高质量的预训练模型进行增量预训练（preprocess.py + pretrain.py），然后进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）
如果计算资源不充裕，可以直接加载高质量的预训练模型，进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）

**Comment by 0x0004** [comment]
> 对于分类任务，如果计算资源比较充裕，可以加载高质量的预训练模型进行增量预训练（preprocess.py + pretrain.py），然后进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）
> 如果计算资源不充裕，可以直接加载高质量的预训练模型，进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）

好的，是不是 增量预训练后微调 效果比 直接微调 要好？

**Comment by Embedding** [comment]
> > 对于分类任务，如果计算资源比较充裕，可以加载高质量的预训练模型进行增量预训练（preprocess.py + pretrain.py），然后进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）
> > 如果计算资源不充裕，可以直接加载高质量的预训练模型，进行微调和推理（finetune/run_classifier.py + inference/run_classifier_infer.py）
> 
> 好的，是不是 增量预训练后微调 效果比 直接微调 要好？

如果增量预训练的语料质量高并且和下游任务的领域相近，一般来说会效果更好
